### ISSUE TITLE
NPCs on kill list oddity.

### ISSUE THREAD

**Issue by DemAvalon** [CleverRaven/Cataclysm-DDA#21793]
### Expected Behavior
When you kill a NPC they appear on your kill list once.
### Actual behavior
When you kill an NPC they appear multiple times or in my case only one NPC you killed appears.
### Steps to reproduce the behavior
Kill many NPCs (migh need to play a little while longer after that).

Images:
https://puu.sh/xrGSN/c9af70b550.png (credit to Astraltor)
https://www.reddit.com/r/cataclysmdda/comments/6u3b7j/the_lady_i_killed_is_haunting_me/

**Comment by Leland** [comment]
Perhaps fallout from #21399

**Comment by lonelychaser** [comment]
One of the ways to replicate this is to kill a NPC, save and exit without closing the application, and then load the game. The NPC will appear twice in the kill list after you load the game.

There seems to be other ways for this to happen as sometimes I saw the kill count pumped up to a absurd value even if I was just exploring tens of map tiles away, even though I am sure that I either had a long session without any saving or I closed the application after save and exit before resuming playing.
